### PR TITLE
Cleaning in tests

### PR DESCRIPTION
**Pull request by bidoubiwa**
Based on these comments the tests are made more readable and consistent: 

- new lines between concerns
- No redundant `Test` in front of test name
- Creation of default contexts where possible 
- remove duplicated tests
- Group related 

<hr>

- https://github.com/meilisearch/instant-meilisearch/pull/707#discussion_r836844862
- https://github.com/meilisearch/instant-meilisearch/pull/707#discussion_r836841423
- https://github.com/meilisearch/instant-meilisearch/pull/707#discussion_r836839633
- https://github.com/meilisearch/instant-meilisearch/pull/707#discussion_r836845760
- https://github.com/meilisearch/instant-meilisearch/pull/707#discussion_r836846982
- https://github.com/meilisearch/instant-meilisearch/pull/707#discussion_r836831810

### CODE REVIEW

File: src/adapter/search-request-adapter/__tests__/search-params.tests.ts [hunk 1]
```diff
@@ -1,211 +1,192 @@
...
+  pagination: { paginationTotalHits: 20, page: 0, hitsPerPage: 6 },
+  defaultFacetDistribution: {},
+}
+
+describe('Parameters adapter', () => {
+  test('adapting a basic searchContext ', () => {
+    const searchParams = adaptSearchParams({
+      ...DEFAULT_CONTEXT,
+      finitePagination: false,
+    })
```
Comment (brunoocasali) [inline src/adapter/search-request-adapter/__tests__/search-params.tests.ts:14]:
```suggestion
    })

```

File: src/adapter/search-request-adapter/__tests__/search-params.tests.ts [hunk 2]
```diff
@@ -1,211 +1,192 @@
...
-
-test('Adapt SearchContext with filters, sort and no geo rules ', () => {
-  const searchParams = adaptSearchParams({
-    indexUid: 'test',
-    pagination: { paginationTotalHits: 20, page: 0, hitsPerPage: 6 },
-    facetFilters: [['genres:Drama', 'genres:Thriller'], ['title:Ariel']],
-    sort: 'id < 1',
-    defaultFacetDistribution: {},
-    finitePagination: false,
+  test('adapting a searchContext with filters and sort', () => {
```
Comment (brunoocasali) [inline src/adapter/search-request-adapter/__tests__/search-params.tests.ts:18]:
```suggestion

  test('adapting a searchContext with filters and sort', () => {
```

**Review by brunoocasali — COMMENTED** [review]
Awesome! I found two things to be changed but everything else is ok!

Two really good references for you about testing good practices are:

- https://www.betterspecs.org/
- http://xunitpatterns.com/Four%20Phase%20Test.html

**Review by brunoocasali — DISMISSED** [review]
Amazing work @bidoubiwa :)

**Review by mdubus — APPROVED** [review]
LGTM 🥳

**Comment by bidoubiwa** [comment]
bors merge

File: src/adapter/search-request-adapter/__tests__/search-params.tests.ts [hunk 3]
```diff
@@ -1,211 +1,194 @@
 import { adaptSearchParams } from '../search-params-adapter'
 
-test('Adapt basic SearchContext ', () => {
-  const searchParams = adaptSearchParams({
-    indexUid: 'test',
-    pagination: { paginationTotalHits: 20, page: 0, hitsPerPage: 6 },
-    defaultFacetDistribution: {},
-    finitePagination: false,
+const DEFAULT_CONTEXT = {
+  indexUid: 'test',
```
Comment (brunoocasali) [inline src/adapter/search-request-adapter/__tests__/search-params.tests.ts:4]: I think the same idea can be applied here, don't you think? https://github.com/meilisearch/meilisearch-ruby/pull/354/files#r934317471

File: src/cache/__tests__/search-cache.tests.ts [hunk 4]
```diff
@@ -39,32 +39,37 @@ describe('Tests on entries in cache', () => {
...
 
-  test('Test to format multiple empty strings', () => {
+  test('to format multiple empty strings', () => {
     const cache = SearchCache()
     const key = cache.formatKey(['', '', ''])
+
     expect(key).toEqual('""""""')
   })
-  test('Test to format empty string', () => {
+  test('to format empty string', () => {
```
Comment (brunoocasali) [inline src/cache/__tests__/search-cache.tests.ts:62]: I think you missed this :)

File: src/adapter/search-request-adapter/__tests__/search-params.tests.ts (hunk 3 above)
Comment (bidoubiwa) [reply src/adapter/search-request-adapter/__tests__/search-params.tests.ts:4]: Do you think `test` is not a good index name? Also these are unit tests so they will not be done against meilisearch

**Pull request merged**